### PR TITLE
Move iOS content validation devicelab test into tool integration test

### PR DESCRIPTION
**Pull request by jmagman**
The archiving was running in devicelab because certs are needed to codesign (see #73577).  However now the certs are available in chromium bots.  Move the archiving test into the existing tool integration test, and delete the devicelab variant.

arm64:
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8757886514651624673/+/u/run_test.dart_for_tool_host_cross_arch_tests_shard_and_subshard_None/test_stdout#L6074_4
x64:
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8757886514651624689/+/u/run_test.dart_for_tool_host_cross_arch_tests_shard_and_subshard_None/test_stdout#L6389_2

Part of https://github.com/flutter/flutter/issues/142070

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/test/host_cross_arch.shard/ios_content_validation_test.dart [hunk 1]
```diff
@@ -354,7 +354,110 @@ void main() {
...
+        projectRoot,
+        'ios',
+        'Runner',
+        'Assets.xcassets',
+        'AppIcon.appiconset',
+        'Icon-App-20x20@1x.png',
+      ));
+      // Resizes app icon to 123x456 (it is supposed to be 20x20).
+      appIconFile.writeAsBytesSync(appIconFile.readAsBytesSync()
+        ..buffer.asByteData().setInt32(16, 123)
```
Comment (christopherfujino) [inline packages/flutter_tools/test/host_cross_arch.shard/ios_content_validation_test.dart:369]: I see you practice the dark arts

File: .ci.yaml [hunk 2]
```diff
@@ -4506,25 +4506,6 @@ targets:
         ["devicelab", "hostonly", "mac", "arm64"]
       task_name: ios_app_with_extensions_test
 
-  - name: Mac_x64_ios ios_content_validation_test
```
Comment (christopherfujino) [inline .ci.yaml:4509]: do we need to test this on both arches?

Comment (christopherfujino) [reply .ci.yaml:4509]: Also, do tool integration tests run on an arbitrary arch?

File: packages/flutter_tools/test/host_cross_arch.shard/ios_content_validation_test.dart (hunk 1 above)
Comment (jmagman) [reply packages/flutter_tools/test/host_cross_arch.shard/ios_content_validation_test.dart:369]: That one was @hellohuanlin #115594

File: .ci.yaml (hunk 2 above)
Comment (jmagman) [reply .ci.yaml:4509]: The tool integration test ios_content_validation_test was already running in  `host_cross_arch.shard` so it will run on both.

**Review by vashworth — APPROVED** [review]
LGTM

File: .ci.yaml (hunk 2 above)
Comment (christopherfujino) [reply .ci.yaml:4509]: oh yeah!

**Pull request merged**